fakestorage/server.go

```diff
@@ -296,12 +299,6 @@ func (s *Server) buildMuxer() {
 	handler.Host(s.publicHost).Path("/{bucketName}").MatcherFunc(matchFormData).Methods(http.MethodPost, http.MethodPut).HandlerFunc(xmlToHTTPHandler(s.insertFormObject))
 	handler.Host(bucketHost).MatcherFunc(matchFormData).Methods(http.MethodPost, http.MethodPut).HandlerFunc(xmlToHTTPHandler(s.insertFormObject))
 
-	// Signed URLs (upload and download)
-	handler.MatcherFunc(s.publicHostMatcher).Path("/{bucketName}/{objectName:.+}").Methods(http.MethodPost, http.MethodPut).HandlerFunc(jsonToHTTPHandler(s.insertObject))
```

Comment:
This can't have ever worked as insertObject expects query parameters encoding the object name, etc...

Reply:
`insertObject` does not necessarily read the object name from query parameters: https://github.com/fsouza/fake-gcs-server/blob/d7832ecd0fab51f1e9047f53375b694aec31e29f/fakestorage/upload.go#L85-L95

I need to take a deeper look at this, but curious if you're basing on comment on the docs? There are many undocumented things in the GCS API that we run into when someone tries to integrate with some different SDK, and I've ironically done a poor job in documenting those in fake-gcs-server 🙈

Reply:
Indeed, insertObject has a peculiar special case that uses `X-Goog-Algorithm` to detect a signed URL and then falls back to implementing something that looks like the XML API. With this PR that can probably be removed :thinking:

Reply:
Yeah, absolutely. insertObject definitely looks like a little monster doing too many things. Thank you very much! I'll go over the PR in details later today.